api/integrations/github/views.py

```diff
@@ -61,6 +62,17 @@ def get_queryset(self):
             organisation_id=self.kwargs["organisation_pk"]
         )
 
+    def create(self, request, *args, **kwargs):
+
+        organisation_id = self.kwargs["organisation_pk"]
+        if GithubConfiguration.has_github_configuration(organisation_id):
+            raise DuplicateGitHubIntegration
+
+        try:
+            return super().create(request, *args, **kwargs)
+        except IntegrityError:
+            raise DuplicateGitHubIntegration
```

Comment:
Is there a way of looking at the `IntegrityError` message in order to raise `DuplicateGitHubIntegration` only when the preconditions have been met?

Reply:
Done